token-allocation/src/adapter.ts

```diff
@@ -36,7 +35,7 @@ export const calculateIndexValue = (index: Index): number => {
   if (!index.every(isPriceSet)) throw new Error('Invalid index: price not set')
   // calculate total value
   return index // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    .reduce((acc, i) => acc.plus(i.units.div(1e18).times(i.price!)), new Decimal(0))
+    .reduce((acc, i) => acc.plus(i.units.times(i.price!)), new Decimal(0))
```

Comment:
You need to divide it here, as we receive `units` on input, not a decimal number of coins.

```suggestion
    .reduce((acc, i) => acc.plus(i.units.div(i.decimal || 1e18).times(i.price!)), new Decimal(0))
```

Reply:
So `token-allocation` adapter won't be responsible of conversion anymore, either aware of decimals. It receives as input the units only, that should be formatted already.

Reply:
The token-allocation adapter should accept base units, and then calculate the price based on a number of base units and defined decimal points for that specific token. If it doesn't know the number of decimal points, because it was not provided, it uses 18 as default.